diff --git a/Source/Generated2/GB_binop__first_fc32.c b/Source/Generated2/GB_binop__first_fc32.c
index 970f661bb5..f742e66774 100644
--- a/Source/Generated2/GB_binop__first_fc32.c
+++ b/Source/Generated2/GB_binop__first_fc32.c
@@ -118,7 +118,7 @@
 
 // disable this operator and use the generic case if these conditions hold
 #define GB_DISABLE \
-    (GxB_NO_FIRST || GxB_NO_FC32 || GxB_NO_FIRST_FC32)
+    (1)
 
 //------------------------------------------------------------------------------
 // C += A+B, all 3 matrices dense
